### ISSUE TITLE
NullReferenceException when getting Location of service in an affinity group 

### ISSUE THREAD

**Issue by ejschaefer** [Azure/classic-iaas-resourcemanager-migration#18]
System.NullReferenceException is thrown when attempting to read Location XML element for services associated with an Affinity Group.

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at MIGAZ.Generator.TemplateGenerator.GenerateTemplate(String tenantId, String subscriptionId, AsmArtefacts artefacts, StreamWriter templateWriter, StreamWriter blobDetailWriter)
   at MIGAZ.Window.btnExport_Click(Object sender, EventArgs e)
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
```

> If the cloud service is associated with an affinity group, the Location element is not returned. 
> https://msdn.microsoft.com/en-us/library/azure/ee460806.aspx

If the service is associated with an affinity group, the Location value is in 

``` xml
    <HostedServiceProperties>
        <ExtendedProperties>
            <ExtendedProperty>
                <Name>ResourceLocation</Name>
                <Value>location-of-resource</Value>
            </ExtendedProperty>
        </ExtendedProperties>
    </HostedServiceProperties>
```

Pull request with fix coming soon

**Comment by PauloRamosGit** [comment]
Thank you for submitting this issue. We are working to include export of cloud services deployed in Affinity Groups. Stay tuned!

**Comment by ejschaefer** [comment]
@PauloRamosGit I have a fix for the subscription that I'm working with using this XPath expression for cloudServiceVM, publicipaddress, and loadbalancer locations

`//HostedServiceProperties/ExtendedProperties/ExtendedProperty[Name='ResourceLocation']/Value`

**Comment by PauloRamosGit** [comment]
The new v1.4.3.0 release is available now, and includes this code change.
Thank you again for the feedback and help.

**Comment by singhkays** [comment]
Fixed with #19. closing.
